compiler_expr.go

```diff
@@ -2086,10 +2088,12 @@ func (c *compiler) compileNumberLiteral(v *ast.NumberLiteral) compiledExpr {
 	}
 	var val Value
 	switch num := v.Value.(type) {
-	case int:
-		val = intToValue(int64(num))
 	case int64:
-		val = int64ToValue(num)
+		if num <= math.MaxInt32 {
+			val = intToValue(num)
+		} else {
+			val = int64ToValue(num)
+		}
```

Comment:
This was necessary because we actually only get int64 values so when we need a `valueInt` it would be treated as `valueInt64` everywhere with some odd behavior (like losing the +/- sign for Infinity).